emperor/support_files/js/controller.js

```diff
@@ -253,10 +254,42 @@ define([
                                  scope.resize(scope.$divId.width(),
                                               scope.$divId.height());
                                }});
+
+    // Set up the context menu
+    this.$contextMenu = $.contextMenu({
+      selector: '.emperor-plot-wrapper',
+      items: {
+        'saveImage': {
+          name: 'Save Image (png)',
+          icon: 'edit',
+          callback: function(key, opts) {
+            scope.screenshot();
+          }
+        }
+      }
+    });
   };
 
   /**
    *
+   * Save the current canvas view to a new window
+   *
+   * @param {string} [type] What type to save the file as. Default png.
+   *
+   **/
+  EmperorController.prototype.screenshot = function(type) {
+    type = type || 'png'
+    // Render all scenes so it's rendered in same context as save
+    for (var i = 0; i < this.sceneViews.length; i++) {
+      this.sceneViews[i].render();
+    }
+    var c = this.renderer.domElement.toDataURL("image/" + type);
+    var w = window.open('about:blank','image from canvas');
```

Comment:
Can you think of a workaround without having to open a new window? We've been discouraged from using direct references to window by the SAGE2 developers, is it possible at all to dump as a file blob, that would also save users 2-3 clicks and a new tab.